### DISCUSSION TITLE
Alternative to `addComputed`?

### DISCUSSION THREAD

**Discussion by arahansen** [pmndrs/valtio#470] [Q&A]
It appears in `valtio@1.6.1` `addComputed` has been deprecated with the following message upon usage:

> addComputed is deprecated. Please consider using `derive` or `proxyWithComputed` instead. Falling back to emulation with derive. 

How do I go about adding a computed value to an existing proxy without needing to create a new proxy?

Here's a pseudo-code example of what I would be trying to do with `addComputed`:

```ts
const state = proxy({})

addComputed(state, { expensiveComputed1: () => { ... } })

// later on

addComputed(state, { expensiveComputed2: () => { ... } })
```

I'm trying to avoid doing something like this:

```ts
let state = proxy({})

state = proxyWithComputed({}, { expensiveComputed1: () => { ... } })

// later on

state = proxyWithComputed({}, { expensiveComputed1: () => { ... }, expensiveComputed2: () => { ... }  })
```

Since this would get more expensive to recompute as more computed values are added

**Comment by dai-shi** [comment]
Hi, you can pass proxy state you want to extend to `derive`.

```js
const state = proxy({})

derive({
  expensiveComputed1: () => { ... },
}, {
  proxy: state
})
```

**Reply by arahansen** [reply]
My understanding though is that `derive` has different semantics than `proxyWithComputed` (and I would assume `addComputed`) in that `proxyWithComputed` uses a snapshot for usage tracking whereas `derive` uses a `get` function to determine changes?

I'm looking to achieve optimization via usage tracking like `proxyWithComputed`. Is that possible?

**Reply by dai-shi** [reply]
`addComputed` didn't have usage tracking, so its semantics is more similar to `derive`.

Note also that, since [v1.6.0](https://github.com/pmndrs/valtio/releases/tag/v1.6.0), `proxyWithComputed` does not include usage tracking and you need to explicitly add it. You could do the same with `derive` combined with `snapshot` (I've never tried it.)
